service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpSendHandler.java

```diff
@@ -331,75 +221,61 @@ public void onLinkFlow(Event event)
                 snd.send(msgData, 0, length);
 
                 snd.advance();
+
+                this.messageToBeSent = null;
```

Comment:
Are multiple messages sent over a single connection? Isn't there a 1:1 connection:operation mapping as in others?

Reply:
There is a 1:1 mapping of amqp connection to c2d message here, yes. By setting the messageToBeSent to null, I am signalling that the message doesn't need to be sent anymore if the ```onLinkFlow()``` event happens again for whatever reason